lib/Onyx.js

```diff
@@ -1248,19 +1259,34 @@ function mergeCollection(collectionKey, collection) {
 
     return getAllKeys()
         .then((persistedKeys) => {
-            // Split to keys that exist in storage and keys that don't
-            const [existingKeys, newKeys] = _.chain(collection)
-                .keys()
-                .partition(key => persistedKeys.includes(key))
-                .value();
+            const existingKeysToKeep = [];
+            const keysToRemove = [];
+            const collectionWithoutKeysToRemove = collection;
+            const keyValuePairsForExistingCollection = [];
+            const keyValuePairsForNewCollection = [];
+            _.each(collection, (value, key) => {
+                if (value === null) {
+                    keysToRemove.push(key);
+                    delete collectionWithoutKeysToRemove[key];
+                    return;
+                }
 
-            const existingKeyCollection = _.pick(collection, existingKeys);
-            const newCollection = _.pick(collection, newKeys);
-            const keyValuePairsForExistingCollection = prepareKeyValuePairsForStorage(existingKeyCollection);
-            const keyValuePairsForNewCollection = prepareKeyValuePairsForStorage(newCollection);
+                if (persistedKeys.includes(key)) {
+                    existingKeysToKeep.push(key);
+                    keyValuePairsForExistingCollection.push([key, removeNullObjectValues(value)]);
+                    return;
+                }
+
+                keyValuePairsForNewCollection.push([key, removeNullObjectValues(value)]);
+            });
 
             const promises = [];
 
+            if (keysToRemove.length > 0) {
+                _.each(keysToRemove, key => cache.drop(key));
+                promises.push(Storage.removeItems(keysToRemove));
```

Comment:
Why not use `Onyx.remove()` here like we did in `Onyx.merge()`?

Reply:
Just because we're bulk-removing items with `Storage.removeItems` instead of multiple calls to `Storage.removeItem`